### ISSUE TITLE
Support serialization of existing nested resource maps

### ISSUE THREAD

**Issue by csjx** [NCEAS/metacatui#623]
While we don't fully support creating and editing of nested data packages yet, we can support editing existing packages that already have nested packages.  In `DataPackage.serialize()` we are removing all package members and then re-aggregating them by iterating through the models and recreating the triple statements.  The issue is that we drop existing nested package triples, removing them from the package.  A second issue is that while we remove the `ore:aggregates` statements with `removeMany()`, we end up not removing any statements where `metadataA cito:documents nestedPackageB`.  Because of this, the `d1_indexer` `ResourceMapSubprocessor` fails to index these resource maps (throwing a `NullPointerException` when it expects both `ore:aggregates` statements when it sees `cito:documents` statements).
Fix `serialize()` to maintain existing nested package entries.

**Comment by csjx** [comment]
The top (parent) resource map version chain (oldest to newest) is:
```
resource_map_urn:uuid:789dbf4b-d859-43f6-b3ea-dbbb1e84a2a4
resource_map_doi:10.18739/A2MP8P
resource_map_doi:10.18739/A26C16
resource_map_urn:uuid:1817d695-2cc6-4bd7-9cba-7e4dbaaa8793
resource_map_urn:uuid:0ea3b16a-4dbe-40c9-86dc-58cd1d32ed89
resource_map_urn:uuid:36a8fbdb-005a-4f51-ac64-d4a1297207f5
resource_map_urn:uuid:23d817e3-bdcc-4257-af55-7d9144e52e05
urn:uuid:9a3f029a-364e-4754-a294-ec731af684a3
```
and the version chain of the science metadata documents associated with the parent package is:
```
urn:uuid:789dbf4b-d859-43f6-b3ea-dbbb1e84a2a4
doi:10.18739/A2MP8P
doi:10.18739/A26C16
urn:uuid:36a8fbdb-005a-4f51-ac64-d4a1297207f5
urn:uuid:23d817e3-bdcc-4257-af55-7d9144e52e05
urn:uuid:e04330f0-6d69-4f7f-8e18-f45bff47bbf8
```
The data team will be manually fixing the nested package, so I just wanted to note which version was problematic.

**Comment by laurenwalker** [comment]
To give some more info on this:

While adding support for nested packages, I found a bug in the package serialization that effects all object types (metadata, data, and nested packages).`documents` relationships were being removed  from the RDF. 

The code flow during serialization was that ALL `documents` relationships were first removed from the RDF, then added back in with the new metadata id. BUT this was not working correctly for all objects, so some `documents` relationships were never added back in, causing a resource map aggregation failure.

This has been fixed, along with other stability changes to the serialization process, in 
a830576.
